doc/api/cli.md

```diff
@@ -2486,6 +2486,34 @@ Any other value will result in colorized output being disabled.
 [`NO_COLOR`][]  is an alias for `NODE_DISABLE_COLORS`. The value of the
 environment variable is arbitrary.
 
+### `NODE_COMPILE_CACHE=dir`
+
+<!-- YAML
+added: REPLACEME
+-->
+
+> Stability: 1.1 - Active Development
+
+When set, whenever Node.js compiles a CommonJS or a ECMAScript Module,
```

Comment:
This description is very technical - I would use more layman terms for the first paragraph.

Reply:
Do you have specific suggestions? I don't want to make it sound too magical to avoid false advertisement.

Reply:
i'll do a follow up PR after this lands to not hold it but ideally something that focuses more on what it does and not the technical implementation